### PR TITLE
[CFM-1211] Addition to the album/photos forms.

### PR DESCRIPTION
**Pull request by ybaras**
https://issuetracker.amplexor.com/jira/browse/CFM-1211

### CODE REVIEW

**Comment by ybaras** [comment]
image

File: project/profiles/capacity4more/modules/c4m/content/c4m_content_photoalbum/c4m_content_photoalbum.module [hunk 1]
```diff
@@ -21,3 +21,29 @@ function c4m_content_photoalbum_c4m_og_vocab_info_content() {
...
+  $form['actions']['add_photos'] = array(
+    '#type' => 'submit',
+    '#value' => t('Add photos'),
+    '#weight' => 5,
+    '#submit' => array('c4m_content_photoalbum_redirect_to_upload_submit'),
+  );
+}
+
+/**
+ * Handles the 'Add photos' button on the node form.
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/content/c4m_content_photoalbum/c4m_content_photoalbum.module]: Use convention.

File: project/profiles/capacity4more/modules/c4m/content/c4m_content_photoalbum/c4m_content_photoalbum.module [hunk 2]
```diff
@@ -21,3 +21,29 @@ function c4m_content_photoalbum_c4m_og_vocab_info_content() {
...
 }
+
+/**
+ * Implements hook_form_FORM_ID_alter().
+ */
+function c4m_content_photoalbum_form_photoalbum_node_form_alter(&$form, &$form_instance, $form_id) {
+  $form['actions']['add_photos'] = array(
+    '#type' => 'submit',
+    '#value' => t('Add photos'),
+    '#weight' => 5,
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/content/c4m_content_photoalbum/c4m_content_photoalbum.module]: Change weight.

File: project/profiles/capacity4more/modules/c4m/content/c4m_content_photoalbum/c4m_content_photoalbum.module [hunk 3]
```diff
@@ -21,3 +21,29 @@ function c4m_content_photoalbum_c4m_og_vocab_info_content() {
     ),
   );
 }
+
+/**
+ * Implements hook_form_FORM_ID_alter().
+ */
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/content/c4m_content_photoalbum/c4m_content_photoalbum.module]: Add docs.

File: project/profiles/capacity4more/modules/c4m/content/c4m_content_photoalbum/c4m_content_photoalbum.module [hunk 4]
```diff
@@ -21,3 +21,29 @@ function c4m_content_photoalbum_c4m_og_vocab_info_content() {
...
+    '#weight' => 5,
+    '#submit' => array('c4m_content_photoalbum_redirect_to_upload_submit'),
+  );
+}
+
+/**
+ * Handles the 'Add photos' button on the node form.
+ * Clicking this button publishes the photo album and redirects to the 'Upload New Items' tab.
+ */
+function c4m_content_photoalbum_redirect_to_upload_submit($form, &$form_state) {
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/content/c4m_content_photoalbum/c4m_content_photoalbum.module]: `c4m_content_photoalbum_add_photos_submit`

File: project/profiles/capacity4more/modules/c4m/content/c4m_content_photoalbum/c4m_content_photoalbum.module [hunk 5]
```diff
@@ -21,3 +21,29 @@ function c4m_content_photoalbum_c4m_og_vocab_info_content() {
...
+  node_form_submit($form, $form_state);
+  if (empty($form_state['nid'])) {
+    return;
+  }
+  $form_state['redirect'] = format_string('node/@nid/upload', array(
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/content/c4m_content_photoalbum/c4m_content_photoalbum.module]: Add line spacing above.

**Comment by ybaras** [comment]
image

File: project/profiles/capacity4more/modules/c4m/content/c4m_content_photoalbum/c4m_content_photoalbum.module [hunk 6]
```diff
@@ -21,3 +21,35 @@ function c4m_content_photoalbum_c4m_og_vocab_info_content() {
...
+
+/**
+ * Form submission handler for `Create Photo album`.
+ *
+ * Handles the 'Add photos' button on the Create Photo album form.
+ * Clicking this button publishes the photo album and redirects to the
+ * 'Upload New Items' tab.
+ */
+function c4m_content_photoalbum_add_photos_submit($form, &$form_state) {
+  node_form_submit($form, $form_state);
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/content/c4m_content_photoalbum/c4m_content_photoalbum.module]: Add a comment that explains the call to this function as well.

**Comment by soniCaH** [comment]
@ybaras can you please re-roll?

File: project/profiles/capacity4more/modules/c4m/content/c4m_content_photoalbum/c4m_content_photoalbum.module [hunk 7]
```diff
@@ -23,6 +23,39 @@ function c4m_content_photoalbum_c4m_og_vocab_info_content() {
 }
 
 /**
+ * Implements hook_form_FORM_ID_alter().
+ *
+ * Adds an `Add Photos` button to the `Create Photo album` form.
+ */
+function c4m_content_photoalbum_form_photoalbum_node_form_alter(&$form, &$form_instance, $form_id) {
+  $form['actions']['add_photos'] = array(
```
Comment (amitaibu) [inline project/profiles/capacity4more/modules/c4m/content/c4m_content_photoalbum/c4m_content_photoalbum.module]: I don't think we need to add a new button, just rename the existing one, and add your own `submit_handler` that will do the redirect.

File: project/profiles/capacity4more/modules/c4m/content/c4m_content_photoalbum/c4m_content_photoalbum.module [hunk 8]
```diff
@@ -23,6 +23,39 @@ function c4m_content_photoalbum_c4m_og_vocab_info_content() {
...
+/**
+ * Form submission handler for `Create Photo album`.
+ *
+ * Handles the 'Add photos' button on the Create Photo album form.
+ * Clicking this button publishes the photo album and redirects to the
+ * 'Upload New Items' tab.
+ */
+function c4m_content_photoalbum_add_photos_submit($form, &$form_state) {
+  // Publish the form by calling the default function.
+  node_form_submit($form, $form_state);
```
Comment (amitaibu) [inline project/profiles/capacity4more/modules/c4m/content/c4m_content_photoalbum/c4m_content_photoalbum.module]: this seems wrong, this submit handler should just be called after the original submit handler was called. and then you can set `$form_state['redirect'] = format_string('node/' . $form_state['nid'] . '/upload'); (i.e. no need for format_string - which should be used only when outputting user generated string)

File: project/profiles/capacity4more/modules/c4m/content/c4m_content_photoalbum/c4m_content_photoalbum.module (hunk 7 above)
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/content/c4m_content_photoalbum/c4m_content_photoalbum.module]: @amitaibu 
It seems to be the request here:
https://issuetracker.amplexor.com/jira/browse/CFM-1211

_cfm-1211__update_photo_and_photoalbum_content_type_-_amplexor

**Comment by amitaibu** [comment]
Can you please attach a gif showing the correct redirection.

**Comment by ybaras** [comment]
.gif of the new button redirection:

test-record

**Pull request merged**